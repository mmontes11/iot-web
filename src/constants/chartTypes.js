export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
