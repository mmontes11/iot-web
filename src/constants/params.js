export const FIRST_PARAM = "FIRST_PARAM";
export const SECOND_PARAM = "SECOND_PARAM";
