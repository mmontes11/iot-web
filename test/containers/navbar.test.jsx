import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { MemoryRouter } from "react-router-dom";
import Navbar from "containers/navbar";
import { initialState } from "../constants/index";

describe("containers/navbar", () => {
  it("renders navbar in initial state", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders navbar in toggled hamburguer menu state", () => {
    const state = {
      ...initialState,
      app: {
        isHamburgerMenuExpanded: true,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates a click in hamburger menu button", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>,
    );

    const hamburgerButton = wrapper.find(".navbar-burger .burger");
    hamburgerButton.first().simulate("click");

    expect(store.getState()).toMatchSnapshot();
  });
  it("simulates a click in logout button", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = mount(
      <MemoryRouter keyLength={0}>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </MemoryRouter>,
    );
    const logoutButton = wrapper.find("#logout-button");
    logoutButton.first().simulate("click");
    expect(store.getState().auth.isAuth).toBeFalsy();
    expect(store.getState()).toMatchSnapshot();
  });
});
