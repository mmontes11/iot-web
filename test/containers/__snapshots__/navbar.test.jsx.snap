// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/navbar renders navbar in initial state and unmounts 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(Connect(ResetOnUnmount)))>
        <Route>
          <Connect(Connect(ResetOnUnmount))
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <Connect(ResetOnUnmount)
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              isHamburgerMenuExpanded={false}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              logout={[Function]}
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              toggleHamburgerMenu={[Function]}
            >
              <ResetOnUnmount
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                isHamburgerMenuExpanded={false}
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                logout={[Function]}
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                reset={[Function]}
                toggleHamburgerMenu={[Function]}
              >
                <Navbar
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  isHamburgerMenuExpanded={false}
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  logout={[Function]}
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  reset={[Function]}
                  toggleHamburgerMenu={[Function]}
                >
                  <nav
                    className="navbar has-shadow is-spaced"
                  >
                    <div
                      className="navbar-brand"
                    >
                      <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="navbar-item"
                        to="/"
                      >
                        <Route
                          path="\\\\/"
                        >
                          <Link
                            aria-current="page"
                            className="navbar-item active"
                            replace={false}
                            style={Object {}}
                            to="/"
                          >
                            <a
                              aria-current="page"
                              className="navbar-item active"
                              href="/"
                              onClick={[Function]}
                              style={Object {}}
                            >
                              <h1
                                className="title"
                              >
                                IoT
                              </h1>
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                      <div
                        className="navbar-burger burger"
                        data-target="navbar-menu"
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        <span />
                        <span />
                        <span />
                      </div>
                    </div>
                    <div
                      className="navbar-menu"
                      id="navbar-menu"
                    >
                      <div
                        className="navbar-start"
                      >
                        <NavLink
                          activeClassName="navbar-item-active"
                          aria-current="page"
                          className="navbar-item"
                          to="/things"
                        >
                          <Route
                            path="\\\\/things"
                          >
                            <Link
                              aria-current={null}
                              className="navbar-item"
                              replace={false}
                              to="/things"
                            >
                              <a
                                aria-current={null}
                                className="navbar-item"
                                href="/things"
                                onClick={[Function]}
                              >
                                <span
                                  className="icon"
                                >
                                  <i
                                    className="fas fa-microchip"
                                  />
                                </span>
                                <span>
                                  Things
                                </span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                        <NavLink
                          activeClassName="navbar-item-active"
                          aria-current="page"
                          className="navbar-item"
                          to="/stats"
                        >
                          <Route
                            path="\\\\/stats"
                          >
                            <Link
                              aria-current={null}
                              className="navbar-item"
                              replace={false}
                              to="/stats"
                            >
                              <a
                                aria-current={null}
                                className="navbar-item"
                                href="/stats"
                                onClick={[Function]}
                              >
                                <span
                                  className="icon"
                                >
                                  <i
                                    className="fas fa-chart-bar"
                                  />
                                </span>
                                <span>
                                  Stats
                                </span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </div>
                      <div
                        className="navbar-end"
                      >
                        <div
                          className="navbar-item"
                        >
                          <div
                            className="field"
                          >
                            <p
                              className="control"
                            >
                              <button
                                className="button is-primary"
                                id="logout-button"
                                onClick={[Function]}
                              >
                                <span
                                  className="icon"
                                >
                                  <i
                                    className="fa fa-user"
                                  />
                                </span>
                                <span>
                                  Logout
                                </span>
                              </button>
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </nav>
                </Navbar>
              </ResetOnUnmount>
            </Connect(ResetOnUnmount)>
          </Connect(Connect(ResetOnUnmount))>
        </Route>
      </withRouter(Connect(Connect(ResetOnUnmount)))>
    </Provider>
  </Router>
</MemoryRouter>
`;

exports[`containers/navbar renders navbar in toggled hamburguer menu state 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(Connect(ResetOnUnmount)))>
        <Route>
          <Connect(Connect(ResetOnUnmount))
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <Connect(ResetOnUnmount)
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              isHamburgerMenuExpanded={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              logout={[Function]}
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              toggleHamburgerMenu={[Function]}
            >
              <ResetOnUnmount
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                isHamburgerMenuExpanded={true}
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                logout={[Function]}
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                reset={[Function]}
                toggleHamburgerMenu={[Function]}
              >
                <Navbar
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  isHamburgerMenuExpanded={true}
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  logout={[Function]}
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  reset={[Function]}
                  toggleHamburgerMenu={[Function]}
                >
                  <nav
                    className="navbar has-shadow is-spaced"
                  >
                    <div
                      className="navbar-brand"
                    >
                      <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="navbar-item"
                        to="/"
                      >
                        <Route
                          path="\\\\/"
                        >
                          <Link
                            aria-current="page"
                            className="navbar-item active"
                            replace={false}
                            style={Object {}}
                            to="/"
                          >
                            <a
                              aria-current="page"
                              className="navbar-item active"
                              href="/"
                              onClick={[Function]}
                              style={Object {}}
                            >
                              <h1
                                className="title"
                              >
                                IoT
                              </h1>
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                      <div
                        className="navbar-burger burger is-active"
                        data-target="navbar-menu"
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        role="button"
                        tabIndex={0}
                      >
                        <span />
                        <span />
                        <span />
                      </div>
                    </div>
                    <div
                      className="navbar-menu is-active"
                      id="navbar-menu"
                    >
                      <div
                        className="navbar-start"
                      >
                        <NavLink
                          activeClassName="navbar-item-active"
                          aria-current="page"
                          className="navbar-item"
                          to="/things"
                        >
                          <Route
                            path="\\\\/things"
                          >
                            <Link
                              aria-current={null}
                              className="navbar-item"
                              replace={false}
                              to="/things"
                            >
                              <a
                                aria-current={null}
                                className="navbar-item"
                                href="/things"
                                onClick={[Function]}
                              >
                                <span
                                  className="icon"
                                >
                                  <i
                                    className="fas fa-microchip"
                                  />
                                </span>
                                <span>
                                  Things
                                </span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                        <NavLink
                          activeClassName="navbar-item-active"
                          aria-current="page"
                          className="navbar-item"
                          to="/stats"
                        >
                          <Route
                            path="\\\\/stats"
                          >
                            <Link
                              aria-current={null}
                              className="navbar-item"
                              replace={false}
                              to="/stats"
                            >
                              <a
                                aria-current={null}
                                className="navbar-item"
                                href="/stats"
                                onClick={[Function]}
                              >
                                <span
                                  className="icon"
                                >
                                  <i
                                    className="fas fa-chart-bar"
                                  />
                                </span>
                                <span>
                                  Stats
                                </span>
                              </a>
                            </Link>
                          </Route>
                        </NavLink>
                      </div>
                      <div
                        className="navbar-end"
                      >
                        <div
                          className="navbar-item"
                        >
                          <div
                            className="field"
                          >
                            <p
                              className="control"
                            >
                              <button
                                className="button is-primary"
                                id="logout-button"
                                onClick={[Function]}
                              >
                                <span
                                  className="icon"
                                >
                                  <i
                                    className="fa fa-user"
                                  />
                                </span>
                                <span>
                                  Logout
                                </span>
                              </button>
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </nav>
                </Navbar>
              </ResetOnUnmount>
            </Connect(ResetOnUnmount)>
          </Connect(Connect(ResetOnUnmount))>
        </Route>
      </withRouter(Connect(Connect(ResetOnUnmount)))>
    </Provider>
  </Router>
</MemoryRouter>
`;

exports[`containers/navbar simulates a click in hamburger menu button 1`] = `
Object {
  "app": Object {
    "isHamburgerMenuExpanded": false,
    "isMapDialogOpened": false,
    "showError": true,
  },
  "auth": Object {
    "isAuth": false,
    "password": null,
    "username": null,
  },
  "request": Object {
    "error": null,
    "pending": 0,
    "statusCode": null,
  },
  "stats": Object {
    "params": Object {
      "observation": Object {
        "isActive": false,
        "isDisabled": true,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
      "reset": Object {
        "isDisabled": true,
      },
      "type": Object {
        "isActive": false,
        "items": Array [
          "event",
          "measurement",
        ],
        "selectedItem": null,
      },
    },
  },
  "things": Object {
    "loadedThings": Array [],
    "selectedThing": null,
  },
}
`;

exports[`containers/navbar simulates a click in logout button 1`] = `
Object {
  "app": Object {
    "isHamburgerMenuExpanded": false,
    "isMapDialogOpened": false,
    "showError": true,
  },
  "auth": Object {
    "isAuth": false,
    "password": null,
    "username": null,
  },
  "request": Object {
    "error": null,
    "pending": 0,
    "statusCode": null,
  },
  "stats": Object {
    "params": Object {
      "observation": Object {
        "isActive": false,
        "isDisabled": true,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
      "reset": Object {
        "isDisabled": true,
      },
      "type": Object {
        "isActive": false,
        "items": Array [
          "event",
          "measurement",
        ],
        "selectedItem": null,
      },
    },
  },
  "things": Object {
    "loadedThings": Array [],
    "selectedThing": null,
  },
}
`;
